addons/mail/models/mail_message.py

```diff
@@ -855,6 +857,19 @@ def _message_format(self, fnames, format_reply=True, legacy=False):
                 'id': message_sudo.author_guest_id.id,
                 'name': message_sudo.author_guest_id.name,
             } if message_sudo.author_guest_id else [('clear',)]
+            # sender: used to display impersonation, mainly for internal users
+            # e.g. when a template is used on behalf of someone else; mail gateway
+            # messages currently consider the author being the sender (as otherwise
+            # it will always be 'Sent by Odoobot')
+            if message_sudo.message_type == 'email':
+                sender = author
+            else:
+                sender = {
+                    'id': message_sudo.create_uid.partner_id.id,
+                    'name': message_sudo.create_uid.name,
```

Comment:
(minor: I assume it is always exactly the same, but we'd want to return partner name in case it was different)